src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryTaskLauncherIntegrationTests.java

```diff
@@ -0,0 +1,257 @@
+	}
+
+	@Test
+	public void testSimpleLaunch() throws InterruptedException {
+
+		String taskId = taskLauncher.asyncLaunch(request).block(Duration.of(5, ChronoUnit.MINUTES));
+
+		System.out.println(">> taskId = " + taskId);
+
+		TaskStatus status = taskLauncher.asyncStatus(taskId).block();
+
+		while (!status.getState().equals(LaunchState.complete)) {
+			System.out.println(">> state = " + status.getState());
+			Thread.sleep(5000);
+			status = taskLauncher.asyncStatus(taskId).block();
+		}
+
+		assertThat(status.getState(), is(LaunchState.complete));
+	}
+
+	@Test
+	public void testSimpleCancel() throws InterruptedException {
+		Map<String, String> envProperties = new HashMap<>();
+		envProperties.put(CloudFoundryDeployerProperties.SERVICES_PROPERTY_KEY, "my_mysql");
+		envProperties.put(CloudFoundryDeployerProperties.MEMORY_PROPERTY_KEY, "1024");
+		envProperties.put(CloudFoundryDeployerProperties.DISK_PROPERTY_KEY, "2048");
+
+		List<String> commandLineArgs = new ArrayList<>(2);
+		commandLineArgs.add("30000");
+
+		request = new AppDeploymentRequest(
+			new AppDefinition("long-runner", Collections.emptyMap()),
+			context.getResource("classpath:long-running-task-1.0.0.BUILD-SNAPSHOT.jar"),
+			envProperties,
+			commandLineArgs);
+
+		String taskId = taskLauncher.asyncLaunch(request).block(Duration.of(5, ChronoUnit.MINUTES));
+
+		System.out.println(">> taskId = " + taskId);
+
+		Thread.sleep(10000L);
+
+		System.out.println(">> About to cancel the task");
+
+		taskLauncher.cancel(taskId);
+
+		TaskStatus status = taskLauncher.asyncStatus(taskId).block();
+
+		while (!status.getState().equals(LaunchState.failed)) {
+			System.out.println(">> state = " + status.getState());
+			Thread.sleep(5000);
+			status = taskLauncher.asyncStatus(taskId).block();
+		}
+
+		assertThat(status.getState(), is(LaunchState.failed));
+	}
+
+	@After
+	public void cleanUp() throws InterruptedException {
+		CloudFoundryClient client = cfAvailable.getResource();
+
+		unbindServices(client, "timestamp");
+		unbindServices(client, "long-runner");
+	}
+
+	private void unbindServices(CloudFoundryClient client, String applicationName) {
+		PaginationUtils.requestClientV3Resources(page -> client.applicationsV3().list(ListApplicationsRequest.builder()
+				.name(applicationName)
+				.page(page)
+				.build()))
+			.log("applicationResponses")
+			.singleOrEmpty()
+			.log("single")
+			.then(applicationResource -> client.serviceBindingsV3().list(ListServiceBindingsRequest.builder()
+				.applicationId(applicationResource.getId())
+				.build()))
+			.log("serviceBindingRequest")
+			.flatMap(serviceBindingsResponse -> Flux.fromIterable(serviceBindingsResponse.getResources()))
+			.log("serviceBindingResponses")
+			.flatMap(serviceBindingResource -> client.serviceBindingsV3().delete(DeleteServiceBindingRequest.builder()
+				.serviceBindingId(serviceBindingResource.getId())
+				.build()))
+			.log("serviceBindingDeletes")
+			.singleOrEmpty()
+			.block();
+	}
+
+	/**
```

Comment:
This is currently unused, but should be when we factor out common logic between stream and tasks TCK